### PR TITLE
Expression Language 6.0.0 Release

### PR DESCRIPTION
**Pull request by markt-asf**
## Specification PR template
When creating a specification project release review, create PRs with the content defined as follows.

Include the following in the PR:
- [X] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [x] Specification PDF in the form of jakarta-wombat-spec-x.y.pdf
- [x] Specification HTML in the form of jakarta-wombat-spec-x.y.html
- [x] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [X] For a [Release Review](https://www.eclipse.org/projects/handbook/#release-review), a summary that a Compatible Implementation is complete, passes the TCK, and that the TCK includes sufficient coverage of the specification. The TCK users guide MUST include the instructions to run the compatible implementations used to validate the release.
Instructions MAY be by reference.
   - [X] Updated release record
   - ~[ ] Generated IP Log~ *no longer required according to project handbook*
   - [X] Email to PMC
   - [X] Start release review by emailing EMO 
- [X] The URL of the OSSRH staging repository for the api, javadoc:
      https://jakarta.oss.sonatype.org/content/repositories/jakartael-1021/
- [X] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee11/staged/eftl/jakarta-expression-language-tck-6.0.0.zip [SHASUM file](https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee11/staged/eftl/jakarta-expression-language-tck-6.0.0-tckinfo.txt)
- [X] The URL of the compatibility certification request issue:
      https://github.com/jakartaee/expression-language/issues/210
- [x] Specification JavaDoc in the wombat/x.y/apidocs directory. 
If desired, an optional second PR can be created to contain just the JavaDoc in the `apidocs` directory.

Note: If any item does not apply, check it and mark N/A below it.

### CODE REVIEW

**Comment by markt-asf** [comment]
Currently blocked waiting for an EL TCK. The current state of the tckrefactor branch is good for the EL TCK (Tomcat 11.0.x SNAPSHOT passes the TCK on Java 17 and Java 21) - we just need a release (staged would be sufficient).

**Comment by alwin-joseph** [comment]
> Currently blocked waiting for an EL TCK. The current state of the tckrefactor branch is good for the EL TCK (Tomcat 11.0.x SNAPSHOT passes the TCK on Java 17 and Java 21) - we just need a release (staged would be sufficient).

I have updated the bundle at https://www.eclipse.org/downloads/download.php?file=/ee4j/jakartaee-tck/jakartaee11/staged/eftl/jakarta-expression-language-tck-6.0.0.zip now using https://ci.eclipse.org/jakartaee-tck/job/11/job/standalone-tck/job/eftl-EL-standalonetck-build/. Can you use this as the staged release or need this to be in some other location?

**Comment by markt-asf** [comment]
I can work with that. Medium to long term, staging into a Maven repo would make consuming the TCK a lot easier but that is not at all required at this point.

**Comment by markt-asf** [comment]
Just to confirm that Tomcat passes the staged TCK. All looks good. Thanks for getting that updated so quickly.
I'm currently testing with a local SNAPSHOT build of Tomcat 11.0.0-M18 to pick up the last minute EL API changes and additions. An 11.0.0-M18 release is likely to be at least a week away so I am currently exploring options for getting a non-local SNAPSHOT build in place that I can reference in the certification request.

**Comment by markt-asf** [comment]
Switching to the CI snapshots for 11.0.0-M18 (which are more recent) triggered a TCK failure for the signature test. Not entirely surprising given the last minutes changes to the EL API but I need to track down where the problem is.

**Comment by markt-asf** [comment]
All good. Just class path issues in my configuration for the sig test. Working on the certification request now.

**Comment by alwin-joseph** [comment]
> All good. Just class path issues in my configuration for the sig test. Working on the certification request now.

Let me refresh the TCK bundle after updating the signature file. Can you please wait for https://ci.eclipse.org/jakartaee-tck/job/11/job/standalone-tck/job/eftl-EL-standalonetck-build/17 to complete.

**Comment by alwin-joseph** [comment]
Can you please retest with the refreshed TCK bundle.

**Comment by markt-asf** [comment]
All good. Still passes.

File: expression-language/6.0/_index.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 title: "Jakarta Expression Language 6.0 (under development)"
```
Comment (Pandrex247) [inline expression-language/6.0/_index.md:2]: The under development bit should be removed with this PR I believe?
```suggestion
title: "Jakarta Expression Language 6.0"
```

File: expression-language/6.0/_index.md [hunk 2]
```diff
@@ -18,29 +18,26 @@ This release makes the dependency on the java.desktop module optional, removes r
...
+    * [Jakarta EE Platform 11 Release Plan](https://jakartaee.github.io/platform/jakartaee11/JakartaEE11ReleasePlan)
+* [Jakarta Expression Language 6.0 Specification Document](./jakarta-expression-language-spec-6.0.pdf) (PDF)
+* [Jakarta Expression Language 6.0 Specification Document](./jakarta-expression-language-spec-6.0.html) (HTML)
+* [Jakarta Expression Language 6.0 Javadoc](./apidocs)
+* [Jakarta Expression Language 6.0 TCK](https://download.eclipse.org/jakartaee/expression-language/6.0/jakarta-expression-language-tck-6.0.0.zip)  ([sig](https://download.eclipse.org/jakartaee/expression-language/6.0/jakarta-expression-language-tck-6.0.0.zip.sig),  [sha](https://download.eclipse.org/jakartaee/expression-language/6.0/jakarta-expression-language-tck-6.0.0.zip.sha256),  [pub](https://raw.githubusercontent.com/jakartaee/specification-committee/master/jakartaee-spec-committee.pub))
 * Maven coordinates
-  * [jakarta.el:jakarta.el-api:jar:5.0.0](https://search.maven.org/artifact/jakarta.el/jakarta.el-api/5.0.0/jar)
+  * [jakarta.el:jakarta.el-api:jar:6.0.0](https://search.maven.org/artifact/jakarta.el/jakarta.el-api/6.0.0/jar)
 -->
```
Comment (Pandrex247) [inline expression-language/6.0/_index.md:33]: Dangling comment close
```suggestion
```

**Review by Pandrex247 — CHANGES_REQUESTED** [review]

File: expression-language/6.0/_index.md [hunk 3]
```diff
@@ -18,29 +18,26 @@ This release makes the dependency on the java.desktop module optional, removes r
...
 
 
 # Compatible Implementations
 
-* Eclipse ExpressLy (TBC)
+* Tomcat 11.0.0-M18-SNAPSHOT (Java 17)
+* Tomcat 11.0.0-M18-SNAPSHOT (Java 21)
```
Comment (Pandrex247) [inline expression-language/6.0/_index.md:39]: You need to link to the download

File: expression-language/6.0/_index.md [hunk 4]
```diff
@@ -18,29 +18,26 @@ This release makes the dependency on the java.desktop module optional, removes r
...
+* ExpressLY (TBD)
```
Comment (Pandrex247) [inline expression-language/6.0/_index.md:40]: If we're doing a release I'd drop this for now.
We can add it back in when it's officially a compatible implementation
```suggestion
```

File: expression-language/6.0/_index.md [hunk 5]
```diff
@@ -18,29 +18,26 @@ This release makes the dependency on the java.desktop module optional, removes r
...
-<!--
-
-The following can be uncommented and version information updated as they become available.
-
-    * [Jakarta EE Platform 10 Release Plan](https://jakartaee.github.io/platform/jakartaee10/JakartaEE10ReleasePlan)
-* [Jakarta Expression Language 5.0 Specification Document](./jakarta-expression-language-spec-5.0.pdf) (PDF)
-* [Jakarta Expression Language 5.0 Specification Document](./jakarta-expression-language-spec-5.0.html) (HTML)
-* [Jakarta Expression Language 5.0 Javadoc](./apidocs)
-* [Jakarta Expression Language 5.0 TCK](https://download.eclipse.org/jakartaee/expression-language/5.0/jakarta-expression-language-tck-5.0.0.zip)  ([sig](https://download.eclipse.org/jakartaee/expression-language/5.0/jakarta-expression-language-tck-5.0.0.zip.sig),  [sha](https://download.eclipse.org/jakartaee/expression-language/5.0/jakarta-expression-language-tck-5.0.0.zip.sha256),  [pub](https://raw.githubusercontent.com/jakartaee/specification-committee/master/jakartaee-spec-committee.pub))
+    * [Jakarta EE Platform 11 Release Plan](https://jakartaee.github.io/platform/jakartaee11/JakartaEE11ReleasePlan)
```
Comment (Pandrex247) [inline expression-language/6.0/_index.md:26]: Also probably worth providing a link to the EL release plan:
```suggestion
    * [Jakarta Expression Language 6.0 Release Plan](https://projects.eclipse.org/projects/ee4j.el/releases/6.0.0/plan)
    * [Jakarta EE Platform 11 Release Plan](https://jakartaee.github.io/platform/jakartaee11/JakartaEE11ReleasePlan)
```

File: expression-language/6.0/_index.md (hunk 3 above)
Comment (markt-asf) [reply expression-language/6.0/_index.md:39]: I'm not a huge fan of using SNAPSHOTs for this. I did it out of necessity. The 11.0.0-M18 release is in progress. I'll update this (and add download links) once the 11.0.0-M18 release completes. I'll note that download links for 11.0.0-M18-SNAPSHOT are in the TCK certification summary.

**Comment by markt-asf** [comment]
The 11.0.0-M18-SNAPSHOT certification request was approved by lazy consensus.
Now the 11.0.0-M18 release is available I have submitted a new certification request based on that release.
I think I am all up to date for the EL 6.0.0 release. Do let me know if there is anything else I need to do.

**Review by Pandrex247 — APPROVED** [review]

**Comment by Pandrex247** [comment]
LGTM, will initiate the ballot 👏

**Comment by Emily-Jiang** [comment]
The Plan Review section does not list much but the delivery has quite a few substantial functions delivered. I suggest listing the following 2 to draw people's attention:

- https://github.com/jakartaee/expression-language/issues/176 Add support for java.util.Optional. The support is not enabled by default. Users must add the new resolver OptionalELResolver to enable this support.

- https://github.com/jakartaee/expression-language/issues/201 Add support for java.lang.Record instances. The support is enabled by default. Support is provided via the new RecordELResolver.

**Comment by arjantijms** [comment]
> The section "**New features, enhancements or additions**" does not list much but the delivery has quite a few substantial functions delivered. I suggest listing the following 2 to draw people's attention:
> 
> * [ELResolver for java.util.Optional expression-language#176](https://github.com/jakartaee/expression-language/issues/176) Add support for java.util.Optional. The support is not enabled by default. Users must add the new resolver OptionalELResolver to enable this support.
> * [Support (and how to support) records in BeanELResolver? expression-language#201](https://github.com/jakartaee/expression-language/issues/201) Add support for java.lang.Record instances. The support is enabled by default. Support is provided via the new RecordELResolver.

Thanks Emily! I agree fully.

**Pull request merged**